File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temp: List[int]) -> List[int]:
        n = len(temp)
        stack = []
        result = [0] * n

        for i, value in enumerate(temp):
            while stack and value > stack[-1][1]:
                stackInd, stackValue  = stack.pop()
                result[stackInd] = i - stackInd
            stack.append((i, value))
        return result

File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] result = new int[n];
        Stack<Integer> stack = new Stack<>();  // Stores index

        for (int i = 0; i < n; i++) {
            int currentTemp = temperatures[i];

            while (!stack.isEmpty() && currentTemp > temperatures[stack.peek()]) {
                int prevIndex = stack.pop();
                result[prevIndex] = i - prevIndex;
            }

            stack.push(i);  // Store index
        }

        return result;
    }
}